oracle/src/oraclewizard.cpp

```diff
@@ -529,3 +544,203 @@ bool SaveSetsPage::validatePage()
+        return false;
+    }
+
+    progressLabel->setText(tr("Downloading (0MB)"));
+    // show an infinite progressbar
+    progressBar->setMaximum(0);
+    progressBar->setMinimum(0);
+    progressBar->setValue(0);
+    progressLabel->show();
+    progressBar->show();
+
+    wizard()->disableButtons();
+    setEnabled(false);
+
+    if(!nam)
+        nam = new QNetworkAccessManager(this);
+    QNetworkReply *reply = nam->get(QNetworkRequest(url));
+
+    connect(reply, SIGNAL(finished()), this, SLOT(actDownloadFinishedTokensFile()));
+    connect(reply, SIGNAL(downloadProgress(qint64, qint64)), this, SLOT(actDownloadProgressTokensFile(qint64, qint64)));
+
+    return false;
+}
+
+void LoadTokensPage::actDownloadProgressTokensFile(qint64 received, qint64 total)
+{
+    if(total > 0)
+    {
+        progressBar->setMaximum(total);
+        progressBar->setValue(received);
+    }
+    progressLabel->setText(tr("Downloading (%1MB)").arg((int) received / 1048576));
+}
+
+void LoadTokensPage::actDownloadFinishedTokensFile()
+{
+    progressLabel->hide();
+    progressBar->hide();
+
+    // check for a reply
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = reply->error();
+    if (errorCode != QNetworkReply::NoError) {
+        QMessageBox::critical(this, tr("Error"), tr("Network error: %1.").arg(reply->errorString()));
+
+        wizard()->enableButtons();
+        setEnabled(true);
+
+        reply->deleteLater();
+        return;
+    }
+
+    // save allsets.json url, but only if the user customized it and download was successfull
+    if(urlLineEdit->text() != QString(TOKENS_URL))
+        wizard()->settings->setValue("tokensurl", urlLineEdit->text());
+    else
+        wizard()->settings->remove("tokensurl");
+
+    wizard()->setTokensData(reply->readAll());
+    reply->deleteLater();
+
+    wizard()->enableButtons();
+    setEnabled(true);
+    progressLabel->hide();
+    progressBar->hide();
+
+    wizard()->next();
+}
+
+SaveTokensPage::SaveTokensPage(QWidget *parent)
+    : OracleWizardPage(parent)
+{
+    defaultPathCheckBox = new QCheckBox(this);
+    defaultPathCheckBox->setChecked(true);
+
+    QGridLayout *layout = new QGridLayout(this);
+    layout->addWidget(defaultPathCheckBox, 0, 0);
+
+    setLayout(layout);
+}
+
+void SaveTokensPage::retranslateUi()
+{
+    setTitle(tr("Tokens imported"));
+    setSubTitle(tr("The tokens has been imported. "
+                   "Press \"Save\" to save the imported tokens to the Cockatrice tokens database."));
+
+    defaultPathCheckBox->setText(tr("Save to the default path (recommended)"));
+}
+
+bool SaveTokensPage::validatePage()
+{
+    bool ok = false;
+    const QString dataDir = 
+#if QT_VERSION < 0x050000
+        QDesktopServices::storageLocation(QDesktopServices::DataLocation);
+#else
+        QStandardPaths::standardLocations(QStandardPaths::DataLocation).first();
```

Comment:
We should make a util function in one location that does the ifdef check internally so we don't need to keep repeating it.

Doesn't need to happen for this pr